tests/unit_and_functional/test_git.py

```diff
@@ -48,74 +48,113 @@ def test_that_git_project_repo_can_be_cloned(
         assert False
 
 
-def test_that_git_project_clone_raises_exceptions():
-    with raises(RuntimeError, match='git clone failed'):
-        download_project_code_from_repo('file:///bad_url')
-
+def test_that_git_project_repo_can_be_cloned_from_github_using_ssh(
+    setup_bodywork_test_project: Iterable[bool],
+    github_repo_connection_string: str,
+    cloned_project_repo_location: Path,
+    set_github_ssh_private_key_env_var: None,
+):
+    try:
+        download_project_code_from_repo(github_repo_connection_string)
+        assert cloned_project_repo_location.exists()
+    except Exception:
+        assert False
+    finally:
+        ssh_dir = Path(".") / SSH_DIR_NAME
+        shutil.rmtree(ssh_dir, ignore_errors=True, onerror=on_error)
 
-@patch('bodywork.git.setup_ssh_for_github')
-def test_that_git_project_clone_returns_git_error_in_exception(mock_setup_ssh: MagicMock): # noqa
-    with raises(RuntimeError, match='fatal: Could not read from remote repository'):
-        download_project_code_from_repo('git@github.com:test/test.git')
 
+def test_that_git_project_repo_can_be_cloned_from_gitlab_using_ssh(
+    setup_bodywork_test_project: Iterable[bool],
+    gitlab_repo_connection_string: str,
+    cloned_project_repo_location: Path,
+    set_gitlab_ssh_private_key_env_var: None,
+):
+    try:
+        ssh_dir = Path(".") / SSH_DIR_NAME
+        os.makedirs(ssh_dir, exist_ok=False)
+        with open(os.path.join(ssh_dir, "id_rsa"), "w"):
+            pass
```

Comment:
`pass`?

Is this a WIP?

Reply:
This is no longer needed, it was required to create a blank file in my previous implementation when the env variable for the private key was removed. 

Another good spot! 🙂